edr_connect/managers/analysis_manager.py

```
@@ -143,16 +149,23 @@ def analyze_by_file(self, alert_info: AlertInfo) -> Optional[Analysis]:
 
         analysis = None
         try:
-            file, zip_password = self.edr_proxy.download_file(alert_info, self.fetched_alerts)
+            file, zip_password, fail_reason = self.edr_proxy.download_file(alert_info, self.fetched_alerts)
             if file:
                 analysis = Analysis(file_stream=file, file_name=f'{alert_info.alert_id}.zip', zip_password=zip_password)
+            elif fail_reason:
+                raise EDRFileDownloadFailureError(fail_reason)
```

I would raise it from s1 proxy instead of return the `fail_reason`, the exception is a better solution